test/test-applications/integrations/TestApplication.StartupHook/Program.cs

```diff
@@ -15,17 +15,37 @@
 // </copyright>
 
 using System.Diagnostics;
+using System.Diagnostics.Metrics;
 using System.Net.Http;
+using CommandLine;
 
 namespace TestApplication.StartupHook
 {
     public class Program
     {
-        private static readonly ActivitySource MyActivitySource = new ActivitySource("TestApplication.StartupHook", "1.0.0");
-
         public static void Main(string[] args)
         {
-            using (var activity = MyActivitySource.StartActivity("SayHello"))
+            var options = Parser.Default.ParseArguments<Options>(args);
```

Comment:
What is the problem if the program would always emit a single span, single metric and single log? 
Do we need to use `CommandLine` and have these `if (o.Trace)` conditionals?
What is the value?

Reply:
I don't have a strong preference for this either. One benefit of the additional complexity with the `CommandLine` options is that it allows us to test additional scenarios, such as "no traces exported when tracing is enabled but no instrumentation is present" (I don't think it's a useful test though).

Reply:
> application in examples folders are used for debugging purpose

Test applications are used for debugging proposes as well (sic!) especially when writing byte-code instrumentation 😉
Examples are intended to be more user-facing - they should serve demonstration purposes.

> It provides a scope for the test to execute what is needed

Currently, the pattern is to create separate test applications. I believe that each "visible" condition makes tests less trustworthy. Mistakes in conditions are often the root cause of bugs. I see the test application is part of our tests. 

Also, I propose to combine the 3 signals into this test application as I would prefer to execute some smoke tests to check if all signals are working as expected. E.g. if one set `*_EXCLUDED_PROCESS` then none of the signals is working.

Reply:
> it allows us to test additional scenarios, such as "no traces exported when tracing is enabled but no instrumentation is present" (I don't think it's a useful test though).

I just cannot come up with a good scenario. Otherwise I would agree 😄

Reply:
As discussed in SIG removed the conditional argument check from test app.